pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -427,6 +436,28 @@ func (vm *volumeManager) WaitForAttachAndMount(ctx context.Context, pod *v1.Pod)
 			return nil
 		}
 
+		if utilfeature.DefaultFeatureGate.Enabled(features.MutableCSINodeAllocatableCount) {
+			for _, volumeToMount := range unattachedVolumeMounts {
+				attachablePlugin, findErr := vm.volumePluginMgr.FindAttachablePluginBySpec(volumeToMount.VolumeSpec)
+				if findErr != nil || attachablePlugin == nil {
+					// This volume type doesn't support the attachable interface, so we can skip our check.
+					continue
+				}
+				if attachablePlugin.VerifyExhaustedResource(volumeToMount.VolumeSpec) {
+					// Return error to the kubelet, which will then trigger the pod termination logic.
+					return &VolumeAttachLimitExceededError{
+						Err: err,
```

Comment:
nit: can you compose err the way it is composed below? So something like:

```
fmt.Errorf(
			"unmounted volumes=%v, unattached volumes=%v, failed to process volumes=%v: %w",
			unmountedVolumes,
			unattachedVolumes,
			volumesNotInDSW,
			err)
```

My point is, user should still be able to see which of the volumes failed to attach and mount.

Reply:
Thank you! @gnufied, updated CR.